src/Sdk/Common/Common/ClientStorage/VssFileStorage.cs

```diff
@@ -192,7 +196,7 @@ internal static string ClientSettingsDirectory
                 // Windows Impersonation is being used.
 
                 // Check to see if we can find the user's local application data directory.
-                string subDir = "GitHub\\ActionsService";
+                string subDir = $"GitHub{c_defaultPathSeparator}ActionsService";
```

Comment:
what about delete `c_defaultPathSeparator` and use `Path.Combine()`? it should do the right thing, I think.

Reply:
That is a better solution, however this constant is later used as the default value of input parameters of several methods, in which paths are formed in different ways (eg via operator + in PathUtil).
I can explore all the different ways of forming paths in project and change them so that in all cases Path.Combine() is used.

Reply:
Can we do something like `string subDir = Path.Combine("GitHub", "ActionsService")`?

Reply:
Of course, but that will require changes in several places. The problem is that the `c_defaultPathSeparator ` is used as an input parameter of one constructor and two methods, so it would be necessary to change their signatures (and all methods and constructors to which the same parameter is passed). Do you want me to change that?

Reply:
Small drive-by comment: if we decide against `Path.Combine` and  end up merging the OS specific `c_defaultPathSeparator` solution, keep in mind that the value of `c_defaultPathSeparator` is used to generate `m_emptyPathSegment` in VssLocalFile, and I'm not sure how that affects the correctness of the invalid path check a few lines later:


https://github.com/ruvceskistefan/runner/blob/3d3388cb394453aa18db46ef50b20ea63fe8a3fd/src/Sdk/Common/Common/ClientStorage/VssFileStorage.cs#L474

```
 m_emptyPathSegment = new string(pathSeparator, 2);
```

https://github.com/ruvceskistefan/runner/blob/3d3388cb394453aa18db46ef50b20ea63fe8a3fd/src/Sdk/Common/Common/ClientStorage/VssFileStorage.cs#L483

```
if (string.IsNullOrEmpty(path) || path[0] != PathSeparator || path.IndexOf(m_emptyPathSegment, StringComparison.Ordinal) >= 0 || (!allowRootPath && path.Length == 1))                
```

Reply:
Okay, makes sense. Then I'll use the `Path.Combine` for this PR instead of OS specific `c_defaultPathSeparator`.